collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -64,7 +69,7 @@ func (r *telemetryAPIReceiver) Start(ctx context.Context, host component.Host) e
 	}()
 
 	telemetryClient := telemetryapi.NewClient(r.logger)
-	_, err := telemetryClient.Subscribe(ctx, r.extensionID, fmt.Sprintf("http://%s/", address))
+	_, err := telemetryClient.Subscribe(ctx, []telemetryapi.EventType{telemetryapi.Platform, telemetryapi.Function, telemetryapi.Extension}, r.extensionID, fmt.Sprintf("http://%s/", address))
```

Comment:
I wonder if it is a good idea to setup a config that users can choose which type of events to subscribe.
Something like: 
```
receivers:
  telemetryapi:
    types: [platform, function, extension]
```
But after I take a look to `config.go`. There is a field extensionID. Is it possible to let `internal/lifecycle/manager` setup the extensionID and users setup the `types` in the collector configuration?

Am I need to fill `extensionID` in config.yaml if

Reply:
@tylerbenson @rapphil @nslaughter Can you please help to advise the above question? Thanks!

Reply:
@jerrytfleung
1. Sorry about the delay in getting you this review. I appreciate the work you're doing on this. Great stuff!
2. I'm also not familiar enough with the extension API to say exactly how that will translate to the config from the top of mind. 
3. The config change that you mention seems to me like it **_will benefit some users and be implemented in a way that's unobjectionable to others_**, but I think these are questions we'd be best to hash out in a Github issue - ideally accompanied by a very focussed PR that includes the proposed implementation.

Reply:
>But after I take a look to config.go. There is a field extensionID. Is it possible to let internal/lifecycle/manager setup the extensionID and users setup the types in the collector configuration

I think we will need to refactor the code a bit so that the extension id is passed to the factory functions `createTracesReceiver` and `createLogsReceiver` through the scope. 

However, if we are adding configuration, we will also need to be able to support multiple instances of the receivers, which does not seem to be the case with the usage of this `SharedComponents` abstraction.